grails-app/services/org/pih/warehouse/invoice/PrepaymentInvoiceService.groovy

```diff
@@ -166,6 +173,9 @@ class PrepaymentInvoiceService {
         OrderItem orderItem = shipmentItem.orderItems?.find { it }
         BigDecimal prepaymentPercent = (orderItem.order.paymentTerm?.prepaymentPercent ?: Constants.DEFAULT_PAYMENT_PERCENT) / 100
         InvoiceItem prepaymentItem = orderItem.invoiceItems.find { it.isPrepaymentInvoice }
+        if (!prepaymentItem) {
```

Comment:
is it even possible to get to this point and there not be a prepaymentItem? Anyways, even if it's impossible, there's nothing wrong with having this additional check.

Reply:
Yes, the case is when you cancel item or adjustment before generating prepayment invoice and the uncanel them. This item is then available to invoice on final invoice but it was not prepaid so we can try to create an inverse